src/StorageSync/StorageSync/ServerEndpoint/RemoveServerEndpointCommandPermission.cs

```diff
@@ -0,0 +1,227 @@
+                        parentResourceName = resourceIdentifier.GetParentResourceName(StorageSyncConstants.SyncGroupTypeName, 0);
+                        storageSyncServiceName = resourceIdentifier.GetParentResourceName(StorageSyncConstants.StorageSyncServiceTypeName, 1);
+                    }
+                    else
+                    {
+                        resourceName = Name;
+                        resourceGroupName = ResourceGroupName;
+                        parentResourceName = SyncGroupName;
+                        storageSyncServiceName = StorageSyncServiceName;
+                    }
+                }
+                Target = string.Join("/", resourceGroupName, storageSyncServiceName, parentResourceName, resourceName);
+                if (ShouldProcess(Target, ActionMessage))
+                {
+                    ServerEndpoint serverEndpoint = StorageSyncClientWrapper.StorageSyncManagementClient.ServerEndpoints.Get(
+                        resourceGroupName,
+                        storageSyncServiceName,
+                        parentResourceName,
+                        resourceName);
+
+                    if(serverEndpoint != null)
+                    {
+                        // Get Registered Server
+                        var serverResourceIdentifier = new ResourceIdentifier(serverEndpoint.ServerResourceId);
+                        StorageSyncModels.RegisteredServer registeredServer = StorageSyncClientWrapper.StorageSyncManagementClient.RegisteredServers.Get(resourceGroupName, storageSyncServiceName, serverResourceIdentifier.ResourceName);
+
+                        if (registeredServer == null)
+                        {
+                            throw new PSArgumentException(StorageSyncResources.MissingServerResourceIdErrorMessage);
+                        }
+
+                        if (!String.IsNullOrEmpty(registeredServer.ApplicationId) && Guid.TryParse(registeredServer.ApplicationId, out Guid serverIdentityGuid))
+                        {
+                            StorageSyncModels.CloudEndpoint cloudEndpoint = StorageSyncClientWrapper.StorageSyncManagementClient.CloudEndpoints
+                            .ListBySyncGroup(resourceGroupName, storageSyncServiceName, parentResourceName).FirstOrDefault();
+
+                            if (cloudEndpoint == null)
+                            {
+                                throw new PSArgumentException(StorageSyncResources.MissingParentResourceIdErrorMessage);
+                            }
+                            var storageAccountResourceIdentifier = new ResourceIdentifier(cloudEndpoint.StorageAccountResourceId);
+                            var scope = $"{cloudEndpoint.StorageAccountResourceId}/fileServices/default/fileshares/{cloudEndpoint.AzureFileShareName}";
+                            bool success = StorageSyncClientWrapper.DeleteRoleAssignmentWithIdentity(storageAccountResourceIdentifier.Subscription,
```

Comment:
same comment about Cluster handling.

Also, is it possible for customers to shoot themselves in the foot with this cmdlet? e.g. they remove permission and suddenly get auth failures? Do we handle that gracefully?

Reply:
Cluster scenario is added.

Please elaborate. SEP is deleted + associated RBAC permission is removed. What auth failures we are referring here and what API(s) for which we are getting auth failures?